picard/cluster.py

```diff
@@ -320,61 +324,31 @@ def cluster(files, threshold, tagger=None):
         for i, file in process_events_iter(enumerate(files)):
             artist = file.metadata["albumartist"] or file.metadata["artist"]
             album = file.metadata["album"]
+
             # Improve clustering from directory structure if no existing tags
             # Only used for grouping and to provide cluster title / artist - not added to file tags.
             if win_compat:
                 filename = ntpath.splitdrive(file.filename)[1]
             else:
                 filename = file.filename
```

Comment:
@zas This is a part I did not touch here, but I wonder about the need for this. If we are on Windows I think `ntpath.splitdrive` should always be done by `album_artist_from_path`, so you don't end up with an artist named "C:".

But do you see any need to have this for other platforms if `config.setting["windows_compatibility"]`  is True? It's really only needed to split this Windows drive letter like "C:\\" or that UNC path to a share, such as "\\\\MyNAS\\theshare\\". But I would not have such paths on e.g. Linux or macOS, and ntpath.splitdrive has no effect on paths like e.g. "/mount/mynas/theshare/xyz".

So I would propose we move the call to `ntpath.splitdrive` to  `album_artist_from_path`, but only check for IS_WIN.

Reply:
Yes, I agree.
But I would do that in another PR.